components/sp-solutions/org.wso2.carbon.sp.solutions.apim.analytics/widgets/APIMApiSubscriberAlertConfiguration/src/APIMApiSubscriberAlertConfiguration.jsx

```diff
@@ -0,0 +1,200 @@
+ *  under the License.
+ *
+ */ 
+
+import React, { Component } from 'react';
+import VizG from 'react-vizgrammar';
+import Widget from '@wso2-dashboards/widget';
+import DropDownMenu from 'material-ui/DropDownMenu';
+import MenuItem from 'material-ui/MenuItem';
+import {MuiThemeProvider} from '@material-ui/core/styles';
+import FlatButton from 'material-ui/FlatButton';
+import TextField from '@material-ui/core/TextField';
+
+class APIMApiSubscriberAlertConfiguration extends Widget {
+    
+    constructor(props) {
+        super(props);
+         
+        this.chartConfig = {
+            "charts": [
+                {
+                    "type": "table",
+                    "columns": [
+                        
+                        {
+                            "name" : "applicationId",
+                            "title": "Application ID"
+                        },
+                        {
+                            "name": "apiName",
+                            "title": "Api Name"
+                        },
+                        {
+                            "name": "apiVersion",
+                            "title": "Api Version"
+                        },
+                        {
+                            "name": "thresholdRequestCountPerMin",
+                            "title": "Threshold Request Count"
+                        }
+                    ]
+                }
+            ],
+            style: {
+                legendTextColor: "#5d6e77"
+            },
+            pagination: true,
+            filterable: true,
+            append: false
+        };
+
+
+        this.metadata = {
+            "names": [
+                "applicationId",
+                "apiName",
+                "apiVersion",
+                "thresholdRequestCountPerMin"
+            ],
+            "types": [
+                "ordinal",
+                "ordinal",
+                "ordinal",
+                "ordinal"
+            ]
+        };
+         
+        this.state = {
+            width: this.props.glContainer.width,
+            height: this.props.glContainer.height,
+            data:[],
+            dataProviderConf:[],
+        };
+        
+        this.props.glContainer.on('resize', () =>
+            this.setState({
+                width: this.props.glContainer.width,
+                height: this.props.glContainer.height
+            })
+        );
+
+        this.handleDataReceived = this.handleDataReceived.bind(this);
+        this.handleDataManaged = this.handleDataManaged.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+    
+    componentDidMount() {
+        super.getWidgetConfiguration(this.props.widgetID)
+            .then((message) => {
+            super.getWidgetChannelManager().subscribeWidget(this.props.id, this.handleDataReceived, message.data.configs.providerConfig);
```

Comment:
use object destructuring for props

Reply:
fix in other places for props and state

Reply:
This was sample code added for revision and removed in 45ea197bf4dd881b1e9d79f5b3ba8d3cc12e05c9 in https://github.com/wso2/analytics-apim